storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerMultipleSubscriptionsTest.java

```diff
@@ -108,7 +114,23 @@ public void testMultiplePartitionSubscriptions() throws Exception {
         final TopicIdPartition followerTopicIdPartition = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition(followerTopic, 0));
         final TopicIdPartition emptyTopicIdPartition = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition(topicWithNoMessages, 0));
 
-        RemoteLogMetadataTopicPartitioner partitioner = new RemoteLogMetadataTopicPartitioner(10) {
+        final RemotePartitionMetadataStore spyRemotePartitionMetadataStore = spy(new RemotePartitionMetadataStore());
+
+        Phaser initializationPhaser = new Phaser(2); // 1 to register test thread, 1 to register leaderTopicIdPartition
+        doAnswer(invocationOnMock -> {
+            Object result = invocationOnMock.callRealMethod();
+            initializationPhaser.arriveAndDeregister();
```

Comment:
Maybe we can use `CountDownLatch` instead? And we can override the stub (i.e create `doAnswer` again) before running `onPartitionLeadershipChanges`? For example:
```java
        CountDownLatch latch = new CountDownLatch(1);
        doAnswer(invocationOnMock -> {
            Object result = invocationOnMock.callRealMethod();
            latch.countDown();
            return result;
        }).when(spyRemotePartitionMetadataStore).markInitialized(any());
...

        latch.await(30_000, TimeUnit.MILLISECONDS);

...

        CountDownLatch latch2 = new CountDownLatch(1);
        doAnswer(invocationOnMock -> {
            Object result = invocationOnMock.callRealMethod();
            latch2.countDown();
            return result;
        }).when(spyRemotePartitionMetadataStore).markInitialized(any());

...
        rlmm().onPartitionLeadershipChanges(Collections.singleton(emptyTopicIdPartition),
            Collections.singleton(followerTopicIdPartition));

```

Reply:
I've added some comments in https://github.com/apache/kafka/pull/15962/commits/9cfda4f3d69049b938da9121dc7ac606b8b04aad to describe the Phaser operations following Kamal's suggestion.

Please let me know if that's okay. I'd prefer not duplicating the stub given a data structure in the standard library fits the purpose but I'm happy to change it if you feel it's confusing for future readers.

Reply:
The comments LGTM. thanks!